File: src/main/java/ru/citeck/idea/codeinsight/forms/components/AssocComponent.java
package ru.citeck.idea.codeinsight.forms.components;

import java.util.Set;

public abstract class AssocComponent extends InputComponent {

    private static final Set<String> ARTIFACT_TYPES = Set.of(
        // TODO: Why alfresco here?
        //AlfrescoModelIndexer.ASSOCIATION,
        //AlfrescoModelIndexer.CHILD_ASSOCIATION
    );

    @Override
    public Set<String> getSupportedArtifactTypes() {
        return ARTIFACT_TYPES;
    }

}

File: src/main/kotlin/ru/citeck/idea/artifacts/action/codeinsight/forms/components/AssocComponent.kt
package ru.citeck.idea.artifacts.action.codeinsight.forms.components

abstract class AssocComponent : InputComponent() {

    companion object {
        private val ARTIFACT_TYPES = setOf<String>()
    }

    override fun getSupportedArtifactTypes(): Set<String> {
        return ARTIFACT_TYPES
    }
}
